lib/src/main/kotlin/com/swmansion/starknet/provider/Provider.kt

```diff
@@ -651,6 +662,19 @@ interface Provider {
      */
     fun getNonce(contractAddress: Felt, blockHash: Felt): Request<Felt>
 
+    /**
+     * Get merkle paths in one of the state tries.
+     *
+     * Get merkle paths in one of the state tries: global state, classes, individual contract.
+     *
+     * @param classHashes list of class hashes for which we want to prove membership
+     * @param contractAddresses list of contract addresses for which we want to prove membership
+     * @param contractsStorageKeys list of contract address and storage keys pairs
+     *
+     * @throws RequestFailedException
+     */
+    fun getStorageProof(classHashes: List<Felt>? = null, contractAddresses: List<Felt>? = null, contractsStorageKeys: List<ContractStorageKey>? = null): Request<StorageProof>
+
```

Comment:
note: It's a bit exceptional situation, due to the fact that creating overloads would cause a conflict, e.g. for
```kotlin
fun getStorageProof(classHashes: List<Felt>, contractsStorageKeys: List<ContractStorageKey>): Request<StorageProof>
```
and
```kotlin
fun getStorageProof(contractAddresses: List<Felt>, contractsStorageKeys: List<ContractStorageKey>): Request<StorageProof>
```
(compiler won't know which method to use, `contractAddresses` and `classHashes` are of the same type). So unfortunately we need to add default values here.